### DISCUSSION TITLE
Explain customer cone

### DISCUSSION THREAD

**Discussion by SichangHe** [SichangHe/internet_route_verification#177] [General]
> In section 5.1.1 it is unclear how the customer cone of an AS is determined?

- [x] #124

**Comment by SichangHe** [comment]
Maybe we rename it to "transitive customers", @cunha?

**Reply by cunha** [reply]
"Customer cone" is synonymous with "transitive customers", but I think sticking to "customer cone" help as it's a well-known concept. CAIDA's CC paper has three different definitions that depend on how peer-peer links are handled, so this may be where the confusion comes from. We should just try to refer to exact version we're using and give the reference:

https://dl.acm.org/doi/10.1145/2504730.2504735

If we have the space, we could include the CC definition we use in the paper.

**Reply by SichangHe** [reply]
> The ppdc-ases files contain the provider-peer customer cones inferred for
each AS.  Each line specifies an AS and all ASes we infer to be reachable
following a customer link.  The format is:
<cone-as> <customer-1-as> <customer-2-as> .. <customer-N-as>

This definition from the data source looks like "transitive customers" to me.

**Comment by cunha** [comment]
"provider-peer customer cones" is the key term. it's referring to a
specific definition in the paper (we should do the same to make it
precise what definition we are using)

**Comment by cunha** [comment]
![image](https://github.com/user-attachments/assets/95d7dc15-1940-461c-aa9a-936bdfd51ca1)

**Comment by SichangHe** [comment]
What if we just use "customers, customers' customers, etc.", @cunha?

**Comment by SichangHe** [comment]
> However, a transit AS X likely intends to export its routes and routes received from any of its customers.

I read the text again. We can implement this exactly as is without going through the customer cone concept. Let's do it that way, @cunha.

**Comment by cunha** [comment]
Just note that it's "routes received from any of its customers", not "routes originated by any of its customers". So we do want the customers -> customers -> customers (like in the customer cone concept). I think just following the customer links (as in your previous message) is what we want.
